### DISCUSSION TITLE
Can't extract color image

### DISCUSSION THREAD

**Discussion by daoth90** [pymupdf/PyMuPDF#1011] [Q&A] [answered]
Hi the community,
I have a pdf file that contains some color and gray images. I don't exactly know well about pix, drawing or the structure of a pdf file (I don't know the format of color image).
I'm using the script from this site but it can extract gray images only (https://www.thepythoncode.com/article/extract-pdf-images-in-python)
I can't attach the file because it's too big (don't know how to split it).
Please help me (debug image frame annot, extract color image or SVG file ...)

**Comment by JorjMcKie** [comment]
Best start with [this](https://github.com/pymupdf/PyMuPDF-Utilities/blob/master/examples/extract-imga.py) script.

**Reply by daoth90** [reply]
Hi @JorjMcKie, today I just tested with the script that you suggested but all the images extracted don't have what I need.

I have updated the pdf file to the driver (https://drive.google.com/file/d/1n1v8cC9B1utcAiWi3tJUHB9mq62GFUJD/view?usp=sharing)

The images I need are on pages 84 and 85 (which is inside the red circle).

Sorry about late reply.

Here are logs when running

> UserWarning: You are running a VERY old version of tkinter 8.5.9
> warnings.warn('You are running a VERY old version of tkinter {}'.format(tclversion_detailed), UserWarning)

> PyMuPDF 1.18.12: Python bindings for the MuPDF 1.18.0 library.
> Version date: 2021-04-10 04:00:00.
> Built for Python 3.8 on darwin (64-bit).
> 
> DEPRECATION WARNING: The system version of Tk is deprecated and may be removed in a future release. Please don't rely on it. Set TK_SILENCE_DEPRECATION=1 to suppress this warning.
> 91 images in total
> 91 images extracted
> total time 13.1148 sec

**Comment by JorjMcKie** [comment, marked as answer]
These are no images, but PDF drawings. So they cannot be extracted via that said script.
Try method `page.get_drawings()` - this extracts drawings in the form of single line, rectangle or curve draw commands.

**Reply by daoth90** [reply]
how do I convert drawing data to image?

**Reply by JorjMcKie** [reply]
Execute the Page / Shape draw commands with arguments extracted from `page.get_drawings()` output. Please read the documentation!

**Reply by JorjMcKie** [reply]
you can also look at the rectangle covered by a drawing and make an appropriate partial pixmap of the page

**Reply by daoth90** [reply]
one more thing, how do I know these items are drawing or anything else or ways to know the type of item I want?

**Reply by JorjMcKie** [reply]
Beforehand? You do not know that. Execute the various extraction types and see if something comes out all and if so what it is.
